materialize/resources/resource_connection.go

```diff
@@ -0,0 +1,838 @@
+	return b
+}
+
+func (b *ConnectionBuilder) ConfluentSchemaRegistrySSLCa(confluentSchemaRegistrySSLCa string) *ConnectionBuilder {
+	b.confluentSchemaRegistrySSLCa = confluentSchemaRegistrySSLCa
+	return b
+}
+
+func (b *ConnectionBuilder) ConfluentSchemaRegistrySSLCert(confluentSchemaRegistrySSLCert string) *ConnectionBuilder {
+	b.confluentSchemaRegistrySSLCert = confluentSchemaRegistrySSLCert
+	return b
+}
+
+func (b *ConnectionBuilder) ConfluentSchemaRegistrySSLKey(confluentSchemaRegistrySSLKey string) *ConnectionBuilder {
+	b.confluentSchemaRegistrySSLKey = confluentSchemaRegistrySSLKey
+	return b
+}
+
+func (b *ConnectionBuilder) ConfluentSchemaRegistrySSHTunnel(confluentSchemaRegistrySSHTunnel string) *ConnectionBuilder {
+	b.confluentSchemaRegistrySSHTunnel = confluentSchemaRegistrySSHTunnel
+	return b
+}
+
+func (b *ConnectionBuilder) ConfluentSchemaRegistryAWSPrivateLink(confluentSchemaRegistryAWSPrivateLink string) *ConnectionBuilder {
+	b.confluentSchemaRegistryAWSPrivateLink = confluentSchemaRegistryAWSPrivateLink
+	return b
+}
+
+func (b *ConnectionBuilder) Create() string {
+	q := strings.Builder{}
+	q.WriteString(fmt.Sprintf(`CREATE CONNECTION %s.%s`, b.schemaName, b.connectionName))
+
+	q.WriteString(fmt.Sprintf(` TO %s (`, b.connectionType))
+
+	if b.connectionType == "SSH TUNNEL" {
+		q.WriteString(fmt.Sprintf(`HOST '%s',`, b.sshHost))
+		q.WriteString(fmt.Sprintf(`USER '%s',`, b.sshUser))
+		q.WriteString(fmt.Sprintf(`PORT %d`, b.sshPort))
+	}
+
+	if b.connectionType == "AWS PRIVATELINK" {
+		q.WriteString(fmt.Sprintf(`SERVICE NAME '%s',`, b.privateLinkServiceName))
+		q.WriteString(fmt.Sprintf(`AVAILABILITY ZONES (%s)`, strings.Join(b.privateLinkAvailabilityZones, ",")))
+	}
+
+	if b.connectionType == "POSTGRES" {
+		q.WriteString(fmt.Sprintf(`HOST '%s',`, b.postgresHost))
+		q.WriteString(fmt.Sprintf(`PORT %d,`, b.postgresPort))
+		q.WriteString(fmt.Sprintf(`USER '%s',`, b.postgresUser))
+		q.WriteString(fmt.Sprintf(`PASSWORD SECRET %s`, b.postgresPassword))
+		if b.postgresSSLMode != "" {
+			q.WriteString(fmt.Sprintf(`, SSL MODE '%s'`, b.postgresSSLMode))
+		}
+		if b.postgresSSHTunnel != "" {
+			q.WriteString(fmt.Sprintf(`, SSH TUNNEL '%s'`, b.postgresSSHTunnel))
+		}
+		if b.postgresSSLCa != "" {
+			q.WriteString(fmt.Sprintf(`, SSL CERTIFICATE AUTHORITY SECRET %s`, b.postgresSSLCa))
+		}
+		if b.postgresSSLCert != "" {
+			q.WriteString(fmt.Sprintf(`, SSL CERTIFICATE SECRET %s`, b.postgresSSLCert))
+		}
+		if b.postgresSSLKey != "" {
+			q.WriteString(fmt.Sprintf(`, SSL KEY SECRET %s`, b.postgresSSLKey))
+		}
+		if b.postgresAWSPrivateLink != "" {
+			q.WriteString(fmt.Sprintf(`, AWS PRIVATELINK %s`, b.postgresAWSPrivateLink))
+		}
+
+		q.WriteString(fmt.Sprintf(`, DATABASE '%s'`, b.postgresDatabase))
+	}
+
+	if b.connectionType == "KAFKA" {
+		if b.kafkaBroker != "" {
+			q.WriteString(fmt.Sprintf(`BROKER '%s',`, b.kafkaBroker))
+		}
+		if b.kafkaBrokers != nil {
```

Comment:
Does this work? I have been checking lists by checking if they are empty or not

Reply:
Just updated this to use the same approach as with tables